src/commands/test/run/xcuitest.ts

```diff
@@ -0,0 +1,74 @@
+import * as iba from "../../../util/misc/ios-bundle-archiver";
+import * as path from "path";
+import * as pfs from "../../../util/misc/promisfied-fs";
+import { CommandArgs, help, longName, hasArg, required } from "../../../util/commandline";
+import { Messages } from "../lib/help-messages";
+import { RunTestsCommand } from "../lib/run-tests-command";
+import { TestCloudError } from "../lib/test-cloud-error";
+import { XCUITestPreparer } from "../lib/xcuitest-preparer";
+
+@help(Messages.TestCloud.Commands.RunXCUITest)
+export default class RunXCUITestCommand extends RunTestsCommand {
+
+  @help(Messages.TestCloud.Arguments.XCUITestBuildDir)
+  @longName("build-dir")
+  @hasArg
+  buildDir: string;
+
+  @help(Messages.TestCloud.Arguments.XCUITestIpaPath)
+  @longName("test-ipa-path")
+  @hasArg
+  testIpaPath: string;
+
+  protected isAppPathRquired = false;
+
+  constructor(args: CommandArgs) {
+    super(args);
+  }
+
+  protected prepareManifest(artifactsDir: string): Promise<string> {
+    let preparer = new XCUITestPreparer(artifactsDir, this.buildDir, this.testIpaPath);
+    return preparer.prepare();
+  }
+
+  protected async validateOptions(): Promise<void> {
+    if (this.buildDir) {
+      if (this.appPath && this.testIpaPath) {
+        throw Error("--build-dir cannot be used when both --app-path and --test-ipa-path are used");
+      }
+      if (!this.appPath) {
+        await this.generateAppIpa();
+      }
+    } else {
+      if (!this.appPath) {
+        throw Error("either --app-path or --build-dir is required");
+      }
+      if (!this.testIpaPath) {
+        throw Error("either --test-ipa-path or --build-dir is required");
+      }
+    }
+  }
+
+  protected getSourceRootDir(): string {
+    return this.buildDir;
+  }
+
+  private async generateAppIpa(): Promise<void> {
+    let appPaths = (await pfs.readdir(this.buildDir)).filter((appPath) => {
+      if (appPath.endsWith("-Runner.app")) {
```

Comment:
In fact, [I can prove it works](http://www.typescriptlang.org/play/#src=%0D%0Alet%20apps%20%3D%20%5B'test.app'%2C%20'test-Runner.app'%5D%3B%0D%0A%0D%0Alet%20appPaths%20%3D%20(apps.filter((appPath)%20%3D%3E%20%7B%0D%0A%20%20%20%20let%20regexp%20%3D%20new%20RegExp(%22%5B%5E-Runner%5D.app%22)%3B%0D%0A%20%20%20%20return%20regexp.test(appPath)%3B%0D%0A%7D))%3B%0D%0A%0D%0Aalert(%22App%20paths%20%3D%20%22%20%2B%20appPaths)%3B);